strawberryfields/engine.py

```diff
@@ -556,21 +552,40 @@ def run(self, program: Program, shots: Optional[int] = None) -> Optional[Result]
             self._connection.cancel_job(job.id)
             return None
 
-    def run_async(self, program: Program, shots: Optional[int] = None) -> Job:
+    def run_async(self, program: Program, *, compile_options=None, **run_options) -> Job:
         """Runs a non-blocking remote job.
 
         In the non-blocking mode, a ``Job`` object is returned immediately, and the user can
         manually refresh the status and check for updated results of the job.
 
         Args:
             program (strawberryfields.Program): the quantum circuit
+            compile_options (None, Dict[str, Any]): keyword arguments for :meth:`.Program.compile`
+
+        Keyword Args:
             shots (Optional[int]): The number of shots for which to run the job. If this
                 argument is not provided, the shots are derived from the given ``program``.
 
         Returns:
             strawberryfields.api.Job: the created remote job
         """
-        return self._connection.create_job(self.target, program, shots)
+        # get the specific chip to submit the program to
+        # TODO: this should be provided by the chip API, rather
+        # than built-in to Strawberry Fields.
+        target = self.DEFAULT_TARGETS.get(self.target, self.target)
+        compile_options = compile_options or {}
+        run_options = run_options.update(self._backend_options)
+
+        if program.target is None or (program.target.split("_")[0] != target.split("_")[0]):
```

Comment:
Seems to be `self._target` in the `Program` class as well.

```suggestion
        if program._target is None or (program._target.split("_")[0] != target.split("_")[0]):
```